src/main/java/io/fabric8/jenkins/openshiftsync/BuildDecisionHandler.java

```diff
@@ -103,21 +107,41 @@ public boolean shouldSchedule(Queue.Task p, List<Action> actions) {
           logger.info("Triggering Jenkins build on WorkflowJob " + wj.getFullName() + " due to running on kubernetes due to " + causeDescription(actions));
           return true;
         }
-        logger.info("Triggering OpenShift Build for WorkflowJob " + wj.getFullName() + " due to " + causeDescription(actions));
-        openShiftClient.buildConfigs()
-          .inNamespace(namespace).withName(jobName)
-          .instantiate(
-            new BuildRequestBuilder()
-              .withNewMetadata().withName(jobName).and()
-              .addNewTriggeredBy().withMessage("Triggered by Jenkins job at " + jobURL).and()
-              .build()
-          );
-        return false;
+
+        /* We are always finding the buildConfig by the the name of JenkinsJobName
+         This is failing in the case of repoName and buildConfigName not same and triggering a build by pushing to master
+         we need to find the buildConfig by gitLabel also as added in launcher backend and
+         we do in PipelineJobListener in case of initial creation
+         */
+        BuildConfig foundBuildConfig = OpenShiftUtils.getBuildConfig(getOpenShiftClient(), namespace, buildConfigname);
+
+        /*In this case if buildConfig name is not same as jenkins jobName converted to buildConfigName
+         we have found the buildConfig by labels and we need to set the buildConfigName equal to Name of the
+         buildConfig we have found to instantiate a build for it
+        */
+        if (foundBuildConfig != null) {
+          buildConfigname = foundBuildConfig.getMetadata().getName();
+          triggerBuildRequest(buildConfigname, namespace, jobURL);
+          return false;
+        }
+
       }
     }
     return true;
   }
 
+  //This will trigger a build for the buildconfig with name buildconfigname
```

Comment:
`triggerBuildRequest(String buildConfigname,....)` is good enough to understand what it does? 
If yes then no need of comment.